src/workspaces/list/WorkspacesListModals.test.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import { goToPath } from 'src/libs/nav';
+import { renderWithAppContexts as render } from 'src/testing/test-utils';
+import { defaultAzureWorkspace, defaultGoogleWorkspace } from 'src/testing/workspace-fixtures';
+import { notifyNewWorkspaceClone } from 'src/workspaces/common/state/useCloningWorkspaceNotifications';
+import { WorkspaceUserActionsContext } from 'src/workspaces/list/WorkspaceUserActions';
+import { NewWorkspaceModalProps } from 'src/workspaces/NewWorkspaceModal//NewWorkspaceModal';
+import { WorkspaceInfo } from 'src/workspaces/utils';
+
+const mockModalFn = jest.fn();
+
+jest.doMock('src/workspaces/NewWorkspaceModal/NewWorkspaceModal', () => ({
+  ...jest.requireActual('src/workspaces/NewWorkspaceModal/NewWorkspaceModal'),
+  NewWorkspaceModal: mockModalFn,
+  default: mockModalFn,
+}));
+
+// this has to be imported with require, and after the mock for NewWorkspaceModal,
+// so the mock is set up when WorkspacesListModals loads its dependencies
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { WorkspacesListModals } = require('src/workspaces/list/WorkspacesListModals');
+
+jest.mock('src/workspaces/common/state/useCloningWorkspaceNotifications', () => ({
+  ...jest.requireActual('src/workspaces/common/state/useCloningWorkspaceNotifications'),
+  notifyNewWorkspaceClone: jest.fn(),
+}));
+
+jest.mock('src/libs/nav', () => ({
+  ...jest.requireActual('src/libs/nav'),
+  goToPath: jest.fn(),
+}));
+
+describe('WorkspacesListModals', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const mockGetWorkspace = jest.fn();
+  const mockRefreshWorkspaces = jest.fn();
+  const mockSetUserActions = jest.fn();
+
+  const props = {
+    getWorkspace: mockGetWorkspace,
+    refreshWorkspaces: mockRefreshWorkspaces,
+  };
+
+  const userActions = {
+    creatingNewWorkspace: false,
+    cloningWorkspace: undefined,
+    deletingWorkspaceId: undefined,
+    lockingWorkspaceId: undefined,
+    sharingWorkspace: undefined,
+    leavingWorkspaceId: undefined,
+    requestingAccessWorkspaceId: undefined,
+  };
+
+  it('renders the NewWorkspaceModal when creating a new workspace', () => {
+    // Arrange
+    mockModalFn.mockImplementation(() => <div data-testid='mockedNewWorkspaceModal' />);
+
+    // Act
+    render(
+      <WorkspaceUserActionsContext.Provider
+        value={{
+          userActions: { ...userActions, creatingNewWorkspace: true },
+          setUserActions: mockSetUserActions,
+        }}
+      >
+        <WorkspacesListModals {...props} />
+      </WorkspaceUserActionsContext.Provider>
+    );
+    // Assert
+    expect(screen.getByTestId('mockedNewWorkspaceModal')).toBeInTheDocument();
+  });
+
+  it('goes to the new workspace when finished', () => {
+    // Arrange
+    mockModalFn.mockImplementation(
+      (props: NewWorkspaceModalProps): React.ReactNode => (
+        // eslint-disable-next-line jsx-a11y/control-has-associated-label, react/button-has-type
```

Comment:
Why not just add a meaningful label, instead of using a testID? Then you could remove the jsx-a11y ignore.